server.h

```diff
@@ -143,22 +149,25 @@ class CTFOServer {
         r("METHOD NOT ALLOWED\n", HTTPResponseCode.MethodNotAllowed);
       } else {
         if (uid == UID::INVALID) {
-          r("NEED VALID UID\n", HTTPResponseCode.BadRequest);
+          r("NEED VALID UID-TOKEN PAIR\n", HTTPResponseCode.BadRequest);
         } else {
-          bool valid_token = false;
-          storage_.Transaction([&uid, &token, &valid_token](StorageAPI::T_DATA data) {
+          bool token_is_valid = false;
+          storage_.Transaction([&uid, &token, &token_is_valid](StorageAPI::T_DATA data) {
                                  const auto auth_token_accessor = Matrix<AuthKeyTokenPair>::Accessor(data);
                                  if (auth_token_accessor.Cols().Has(token)) {
                                    // Something went terribly wrong
-                                   // if we have more than one `device_id` + `app_key` pair for token.
+                                   // if we have more than one authentication key for token.
                                    assert(auth_token_accessor[token].size() == 1);
                                    if (auth_token_accessor[token].begin()->valid) {
-                                     valid_token = true;
+                                     // Double check, if the provided `uid` is correct as well.
+                                     const auto auth_uid_accessor = Matrix<AuthKeyUIDPair>::Accessor(data);
+                                     token_is_valid =
```

Comment:
Or you can return it from the lambda, and have `const bool token_is_valid = storage_.Transaction(...).Go();`, right?